winml/adapter/WinMLAdapter.cpp

```diff
@@ -571,6 +568,9 @@ class WinMLAdapter : public Microsoft::WRL::RuntimeClass<
   // registered schema are reachable only after upstream schema have been revised in a later OS release,
   // which would be a compatibility risk.
   HRESULT STDMETHODCALLTYPE OverrideSchemaInferenceFunctions() override try {
+  // lazy load the lotus_environment. This is the first place where WinML requires the LotusEnvironment to be initialized.
+    lotus_environment_ = std::shared_ptr<WinML::LotusEnvironment>(
```

Comment:
`std::make_shared(PheonixSingleton<WinML::LotusEnvironment>());` should work the same

Reply:
I just looked up the definition for PheonixSingleton and it returns std::shared_ptr<T> by value, is there a reason not to write `lotus_environment_ = PheonixSingleton<WinML::LotusEnvironment>();`?

Reply:
good point! Originally I was just preserving what the constructor of WinMLAdapter was doing: lotus_environment_(PheonixSingleton<WinML::LotusEnvironment>() which calls the constructor of std::shared_ptr<T>.

I think it's fine to do this: lotus_environment_ = PheonixSingleton<WinML::LotusEnvironment>();

---
In reply to: [354569655](https://github.com/microsoft/onnxruntime/pull/2562#discussion_r354569655) [](ancestors = 354569655)